core/server/common/src/main/java/alluxio/master/journal/raft/transport/CopycatGrpcClient.java

```diff
@@ -101,29 +90,14 @@ public CopycatGrpcClient(AlluxioConfiguration conf, UserState userState) {
   }
 
   @Override
-  public synchronized CompletableFuture<Void> close() {
-    if (mClosed) {
-      return CompletableFuture.completedFuture(null);
-    }
-
-    LOG.debug("Closing copycat transport client with {} gRPC channels.", mChannels.size());
+  public CompletableFuture<Void> close() {
+    LOG.debug("Closing copycat transport client with {} connections.", mConnections.size());
     // Close created connections.
     List<CompletableFuture<Void>> connectionCloseFutures = new ArrayList<>(mConnections.size());
-    for (Connection connection : mConnections) {
-      connectionCloseFutures.add(connection.close());
+    for (Connection connectionPair : mConnections) {
+      connectionCloseFutures.add(connectionPair.close());
```

Comment:
Why is this a `connectionPair` now?

Reply:
Because of 1-1 mapping, I don't need separate collections but could put them alongside.